### ISSUE TITLE
Build in subdirs currently serial, could be parallel

### ISSUE THREAD

**Issue by mohawk2** [Perl-Toolchain-Gang/ExtUtils-MakeMaker#207]
Currently eg `subdirs_dynamic` has a list of recipes each of which is equivalent to `cd dirX && make dynamic`. This forces serial building. Instead it could be done in a parallelisable way.

It has been suggested the way forward is to have one new target per subdir, which might need minor name-mangling, together with the target-type, so eg `subdir_dirX_dynamic` for the above.

@shadowcat-mst further suggested, and I agree, that using a WriteMakefile param to switch this on will avoid back-compat problems.

**Comment by mohawk2** [comment]
cf https://github.com/PDLPorters/pdl/commit/907e130d34b69789831f7c682bf0ce8e31ec9f95

**Comment by Leont** [comment]
This has a risk of process stampede: if the parent make is -j9 and it has five subdirs that all run with -j9, you suddenly got a whole lot of subprocesses. Without scheduling between the different make incantations, this may not be beneficial.

**Comment by mohawk2** [comment]
That is true. However, I believe that GNU make does coordinate between its various children.

**Comment by mohawk2** [comment]
The real, scalable solution to this is to not use subdirs, but `XSMULTI`. PDL 2.096+ does this, and parallel builds work fine.
